internal/resources/policy/kind/network/resource/network_policy_mock_test.go

```diff
@@ -0,0 +1,300 @@
+			OrgID: testConfig.ScopeHelperResources.OrgID,
+		},
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: nil,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID: "workspace1",
+		},
+	}
+
+	postWorkspaceResponseModel := &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceWorkspace{
+		FullName: &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName{
+			Name:  testConfig.ScopeHelperResources.Workspace.Name,
+			OrgID: testConfig.ScopeHelperResources.OrgID,
+		},
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: nil,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID: "workspace1",
+		},
+	}
+
+	postRequest := &policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicyRequest{
+		Policy: postRequestModel,
+	}
+
+	postResponse := &policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicyResponse{
+		Policy: postResponseModel,
+	}
+
+	postWorkspaceRequest := &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceRequest{
+		Workspace: postWorkspaceRequestModel,
+	}
+
+	postWorkspaceResponse := &workspacemodel.VmwareTanzuManageV1alphaWorkspaceResponse{
+		Workspace: postWorkspaceResponseModel,
+	}
+
+	// GET Network Policy mock setup
+	getModel := &policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicy{
+		FullName: &policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyFullName{
+			Name:          testConfig.NetworkPolicyName,
+			OrgID:         testConfig.ScopeHelperResources.OrgID,
+			WorkspaceName: testConfig.ScopeHelperResources.Workspace.Name,
+		},
+		Spec: &policySpec,
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: referenceArray,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID:             "policy1",
+			ResourceVersion: "v1",
+		},
+	}
+
+	getWorkspaceModel := &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceWorkspace{
+		FullName: &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName{
+			Name:  testConfig.ScopeHelperResources.Workspace.Name,
+			OrgID: testConfig.ScopeHelperResources.OrgID,
+		},
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: nil,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID: "workspace1",
+		},
+	}
+
+	getResponse := &policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyGetPolicyResponse{
+		Policy: getModel,
+	}
+
+	getWorkspaceResponse := &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceGetWorkspaceResponse{
+		Workspace: getWorkspaceModel,
+	}
+
+	// Setup HTTP Responders
+	postWorkspaceEndpoint := fmt.Sprintf("https://%s/v1alpha1/workspaces", endpoint)
```

Comment:
Can you write a generic helper function to return endpoints like how it is done in the clients?